spec/worker_spec.rb

```diff
@@ -0,0 +1,139 @@
+RSpec.describe ActiveAdmin::AsyncExporter::Worker do
+  let!(:users) do
+    %w[user_1 user_2 user_3].map do |username|
+      User.find_or_create_by!(username: username)
+    end
+  end
+  let(:controller) { Admin::UsersController.new }
+  let(:decorate_model) { false }
+  let(:file_name) { 'file_name' }
+  let(:query) { nil }
+  let(:csv_path) { './csv' }
+  let(:admin) { User.find_or_create_by!(username: 'AdminUser') }
+  let(:csv_fixture_file_path) { "./spec/fixtures/files/#{csv_fixture_file_name}.csv" }
+  let(:csv_fixture_file_name) { 'username' }
+
+  let(:admin_report) do
+    AdminReport.create!(author_id: admin.id,
+                        entity: User.name,
+                        status: :pending)
+  end
+
+  let(:columns) { { username: { block: false, value: 'username' } } }
+
+  let(:options) do
+    {
+      admin_report_id: admin_report.id,
+      controller: controller.class.name,
+      columns: columns,
+      decorate_model: decorate_model,
+      file_name: file_name,
+      query: query
+    }
+  end
+
+  subject do
+    described_class.perform_now(options)
+  end
+
+  context 'with valid options' do
+    let(:test_date) { Time.current }
+    let(:full_file_name) { [file_name, test_date.to_i].join('_') + '.csv' }
+    let(:file_path) { "#{csv_path}/#{full_file_name}" }
+
+    before do
+      Timecop.freeze(test_date)
+    end
+
+    after do
+      FileUtils.rm(file_path)
+      Timecop.return
+    end
+
+    shared_examples 'creates the physical file with data' do
+      it 'creates the physical file at the given location_url' do
+        subject
```

Comment:
maybe it's a good idea to add `expect(File.open(admin_report.reload.location_url)).to_not be` before running the `subject` to make sure that the subject is actually creating it. Same for the other tests

Reply:
Cannot do `File.open` of a file that does not exist. I can try to do something like `expect { File.open(admin_report.reload.location_url.to_s) }.to raise_exception(Errno::ENOENT)`, WDYT?